response.go

```diff
@@ -88,55 +89,41 @@ func (res *Response) Encoding() (encoding.Encoding, error) {
 
 // DetectContentType detects and sets the response content type and charset.
 func (res *Response) DetectContentType(ct string) {
-	// Sensible defaults
-	res.MediaType = "text/plain"
-	res.Charset = ""
-
 	// Autodetect if not passed a Content-Type header
 	if ct == "" {
 		ct = http.DetectContentType(res.Body)
 	}
 
 	// Content type (e.g. text/plain or text/html)
-	mt, params, err := mime.ParseMediaType(ct)
+	mt, _, err := mime.ParseMediaType(ct)
 	if err != nil {
 		return
 	}
 	res.MediaType = mt
```

Comment:
I'm uncomfortable at this existing code, since `http.DetectContentType` is not constrained to text and will fall back to `application/octet-stream`.  We should at least be detecting `application/octet-stream` and treating it as "nothing found" and instead persisting with _our_ default.